src/cargo/util/toml/mod.rs

```diff
@@ -1909,7 +1910,28 @@ impl<P: ResolveToPath + Clone> schema::TomlDetailedDependency<P> {
                 // always end up hashing to the same value no matter where it's
                 // built from.
                 if cx.source_id.is_path() {
-                    let path = cx.root.join(path);
```

Comment:
Question for reviewers: on line 1903 we're still adding the path to `nested_paths` - is this going to be an issue for paths with a base?

I've tried tracing where that flows to, but it seems to be returned and renamed by a bunch of functions.

Reply:
FYI at least for myself, I'm not planning on looking at this until the RFC is approved.  So unsure how important it is to figure out these details and fully polish it up atm

Reply:
Looks like `nested_paths` did need to be updated: I've added a test and fixed the code.